ts/components/wallet/card/CardComponent.tsx

```diff
@@ -194,32 +192,55 @@ export default class CardComponent extends React.Component<Props> {
     if (type === "Preview") {
       return null;
     }
-    // Right icon, basically needed for the sole "Header" variant
+
+    // Extract the brand name from the credit card pan
+    const detectedBrand: SupportedBrand = CreditCardDetector.validate(
+      some(creditCard.pan)
+    );
+    /**
+     * Extract the brand logo from the brand name
+     * since the keys of the @link{cardIcons} are the @link{CreditCardType}
+     * we must manage the case with the different name but the same logo
+     */
+    const creditCardType = CreditCardType.decode(
+      detectedBrand.name.toUpperCase()
+    );
+    const logo = creditCardType.fold(
```

Comment:
We should have a single point of truth on those stuff that are cross in the app domani.
Could it be a better solution add a new brand name in the relative mapping?
![Schermata 2022-01-25 alle 16 32 03](https://user-images.githubusercontent.com/822471/151007617-105f39d4-4940-47f6-87f3-f987b7dd5d7e.png)
It will raise an error if you don't associate the relative `cardIcons` mapping

Reply:
Thank you good idea! I accidentally thought the CrediCardType was generated. This is the reason why I choose another way! 
Refactored in [1d62487](https://github.com/pagopa/io-app/pull/3691/commits/1d624875c481ac4a2558465739d268483ec6902f)